test_plans/kernel_bundle.asciidoc

```diff
@@ -0,0 +1,139 @@
+** `true` for a kernel decorated with `[[sycl::device_has(aspect::fp64)]]` if and only if the selected device has that aspect.
+** `true` for a kernel decorated with `[[sycl::device_has(aspect::atomic64)]]` if and only if the selected device has that aspect.
+
+* for multiple kernels result is equal to AND operation applied to the multiple calls - each one using single kernel id
+** Check with vector with built-in kernel ids and with the selected device - should return `true`
+** To previous vector add kernel id with large `[[sycl::reqd_work_group_size(4294967295)]]` attribute and with the selected device - should return `false` if value reported by `info::device::max_work_group_size` is less than 4294967295.
+
+
+==== Tests for `template<typename KernelName> bool is_compatible(const device &dev)`
+
+Same tests as previous section except no zero, built-in and multiple kernels cases.
+
+=== `sycl::kernel_bundle` API
+
+==== Deleted default constructor
+
+Check that `std::is_default_constructible_v<sycl::kernel_bundle>` is `false`.
+
+==== Test for `get_backend()`
+
+Call function `kernel_bundle::get_backend()` and check that return type is `sycl::backend`.
+
+==== Tests for `get_devices()`
+
+* Create a 1-element `std::vector<device>` containing the selected device.
+* Create kernel_bundle via `get_kernel_bundle<bundle_state::executable>(const context&, const std::vector<device>&)`.
+* Call `get_devices()`.
+* Check that return type is std::vector<device>.
+* Check that return vector contains only selected device.
+
+==== `has_kernel(const kernel_id &kernelId, const device &dev)`
+
+* Create a kernel bundle in executable state which contains all the application's kernels by calling `get_kernel_bundle<bundle_state::executable>(const context &)`.
+* Iterate over all of the kernel ids returned by `get_kernel_ids()`.
+* Verify that `has_kernel(kernelId, dev)` returns `true` if and only if `is_compatible({kernelId}, const device&)` for selected device is `true`.
+
+==== `std::vector<kernel_id> get_kernel_ids()`
+
+Define 4 kernels.
+Get kernel_bundle that contains all of the kernels defined in the application via `get_kernel_bundle<bundle_state::executable>(const context &)`
+Check that result of `get_kernel_ids`:
+
+* has type `std::vector<kernel_id>`
+* has size at least 4
+* has only different kernel ids
+
+==== `get_kernel()`
+
+* Define kernel KernelName.
+* Get a kernel bundle in executable state that contains this kernel via `get_kernel_bundle<KernelName, bundle_state::executable>(const context &)`.
+* Use `get_kernel<KernelName>()` to get kernel and check that return type is `sycl::kernel`.
+
+=== Tests for working with specialization constants
+
+Partially tested in https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/tests/specialization_constants/specialization_constants_via_kernel_bundle.h
+
+There are should be defined spec constant `SpecName`.
```

Comment:
```suggestion
There are two spec constant defined: `SpecName` and `OtherSpecName`.
```

The test below says that there is also a spec constant named `OtherSpecName`.

Reply:
Returned `OtherSpecName`.